stream_alert/shared/alert.py

```diff
@@ -81,12 +91,15 @@ def __init__(self, rule_name, record, outputs, **kwargs):
         self.attempts = int(kwargs.get('attempts', 0)) or 0  # Convert possible Decimal to int
         self.cluster = kwargs.get('cluster') or None
         self.context = kwargs.get('context') or {}
-        self.dispatched = int(kwargs.get('dispatched', 0)) or 0
+
+        # datetime.min isn't supported by strftime, so use Unix epoch instead for default value
+        self.dispatched = kwargs.get('dispatched') or datetime(year=1970, month=1, day=1)
```

Comment:
tip: you can also use `time.gmtime(0)` to accomplish this. but not a necessary change since `time` is not imported here

Reply:
Good idea! Unfortunately, `time.gmtime(0)` returns a `time.struct_time`, which isn't compatible with `timedelta`. So it would have to be explicitly converted to a `datetime` object, like so:

```python
dispatched = datetime.fromtimestamp(time.mktime(time.gmtime(0)))
```

My thought was it was just easier to keep everything as `datetime` and `timedelta` (both members of the `datetime` class)